fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/RegistrySyncManager.java

```diff
@@ -130,13 +133,44 @@ public static CompoundTag toTag(boolean isClientSync) {
 				}
 			}
 
-			if (REGISTRY_BLACKLIST.contains(registryId)) {
+			CompoundTag existingRegistryData = null;
+
+			if (activeIdMap != null && activeIdMap.contains(registryId.toString())) {
+				existingRegistryData = activeIdMap.getCompound(registryId.toString());
+			}
+
+			RegistryAttributeHolder attributeHolder = RegistryAttributeHolder.get(registry);
+
+			if (!isClientSync && !attributeHolder.hasAttribute(RegistryAttribute.PERSISTED)) {
+				LOGGER.debug("Not saving non-persistent registry: " + registryId);
 				continue;
-			} else if (isClientSync && REGISTRY_BLACKLIST_NETWORK.contains(registryId)) {
+			}
+
+			if (isClientSync && !attributeHolder.hasAttribute(RegistryAttribute.SYNCED)) {
+				LOGGER.debug("Not syncing registry: " + registryId);
 				continue;
 			}
```

Comment:
IMO this code would be clearer by writing it like thhs since I had to look twice whether it was exhaustive:

```
if (!attributeHolder.hasAttribute(isClientSync ? RegistryAttribute.SYNCED : RegistryAttribute.PERSISTED)) {
	LOGGER.debug("Not {} registry: {}", isClientSync ? "syncing" : "saving", registryId);
	continue;
}
```